### PR TITLE
Rename bus peripheral classes, fields and references to communications

### PR DESCRIPTION
**Pull request by adrianstevens**

### CODE REVIEW

File: Source/Meadow.Foundation.Core/Communications/ByteCommsSensorBase.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ public abstract class ByteCommsSensorBase<UNIT> :
         PollingSensorBase<UNIT>, IDisposable where UNIT : struct
     {
         /// <summary>
-        /// Peripheral object, i.e. an I2CPeripheral or SpiPeripheral
+        /// Bus communications object, i.e. an I2cCommunications or SpiCommunications
         /// </summary>
         protected IByteCommunications? Peripheral { get; set; }
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Core/Communications/ByteCommsSensorBase.cs:17]: Should this still be called Peripheral?

File: Source/Meadow.Foundation.Peripherals/RTCs.Ds323x/Driver/Ds323x.cs [hunk 2]
```diff
@@ -107,7 +107,7 @@ protected Ds323x(I2cPeripheral peripheral, IPin interruptPin)
         /// <summary>
         /// Create a new Ds323x object
         /// </summary>
-        protected Ds323x(I2cPeripheral peripheral, IDigitalInputPort interruptPort)
+        protected Ds323x(I2cCommunications peripheral, IDigitalInputPort interruptPort)
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Peripherals/RTCs.Ds323x/Driver/Ds323x.cs:110]: Rename here, correct?

File: Source/Meadow.Foundation.Peripherals/RTCs.Ds323x/Driver/Ds323x.cs [hunk 3]
```diff
@@ -89,7 +89,7 @@ static class Registers
         /// <summary>
         /// Create a new Ds323x object
         /// </summary>
-        protected Ds323x(I2cPeripheral peripheral, IPin interruptPin)
+        protected Ds323x(I2cCommunications peripheral, IPin interruptPin)
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Peripherals/RTCs.Ds323x/Driver/Ds323x.cs:92]: Rename here, correct? peripheral => I2cComms

File: Source/Meadow.Foundation.Peripherals/Sensors.Motion.Adxl3xx/Driver/Drivers/Adxl362.cs [hunk 4]
```diff
@@ -70,8 +70,8 @@ public partial class Adxl362
         /// </summary>
         public Frequency SpiBusSpeed
         {
-            get => ((SpiPeripheral)Peripheral).BusSpeed;
-            set => ((SpiPeripheral)Peripheral).BusSpeed = value;
+            get => ((SpiCommunications)Peripheral).BusSpeed;
```
Comment (jorgedevs) [inline Source/Meadow.Foundation.Peripherals/Sensors.Motion.Adxl3xx/Driver/Drivers/Adxl362.cs:73]: Peripheral should be SpiComms, no?

File: Source/Meadow.Foundation.Core/Communications/ByteCommsSensorBase.cs (hunk 1 above)
Comment (adrianstevens) [reply Source/Meadow.Foundation.Core/Communications/ByteCommsSensorBase.cs:17]: great call - I'm on it - I'm thinking `ByteComms`

File: Source/Meadow.Foundation.Peripherals/RTCs.Ds323x/Driver/Ds323x.cs (hunk 2 above)
Comment (adrianstevens) [reply Source/Meadow.Foundation.Peripherals/RTCs.Ds323x/Driver/Ds323x.cs:110]: yup :)

File: Source/Meadow.Foundation.Peripherals/RTCs.Ds323x/Driver/Ds323x.cs (hunk 3 above)
Comment (jorgedevs) [reply Source/Meadow.Foundation.Peripherals/RTCs.Ds323x/Driver/Ds323x.cs:91]: @adrianstevens I think this one is still missing

**Pull request merged**